Create table: Split insertion into multiple batches if more than a limiting number of rows are to be inserted

garland-culbreth opened the pull request:
# Proposed changes

In `devel/table/create_table.cpp`, if the number of rows being inserted is greater than a limiting number, split the insertion into multiple batches rather than inserting all of them at once.

# Why is this needed

In situations where there are a large number of grid points in age/time, a large number of covariates, and a large number of samples, it can happen that the number of rows being inserted into the variable table in `dismod.db` is too large for sqlite's maximum string length ([sqlite limit 1](https://www.sqlite.org/limits.html)). Splitting the statement up into multiple avoids this problem.

# Tests

With this change, `bin/check_all.sh` passes all tests. Creating a new test which inserts more than the limiting number of rows to ensure the split operates correctly is TODO.

# 2DO

- [x] Add a new test which inserts a very large number of rows to ensure split insertions operate properly.

garland-culbreth committed:
split_table_insert: limit max insertion rows

set a max on the nunmber of rows to be inserted by the command, and
repeat the command as many times as required to cover n_row.

bradbell commented:
If you add a test with a large number of rows, it might be good if :

1. You make this an example (so it has documentation)
2. The model is linear (so that asymptotic statistics are correct); e.g. a direct measurement of Sincidence where iota is the only variable and is constant w.,r.t. age, time.
3.  Check the standard deviation  of the esiitmate as a function of the number of data points and measurement std.
4. Make the data point mean and standard deviation the same for all data points global variables at the top that can be changed.

garland-culbreth commented:
Edited the opening post to be 2DO to be consistent with the convention.

For the test, I'm hoping I can make `cut_size` an optional function parameter, using 5000000 as the default value; that way a test could still be done with a small/typical amount of data and `cut_size` set very small. I haven't gotten that together yet though

bradbell commented:
I think that a much smaller value is reasonable.; say 1,000 rows.  I suspect that creating such a large sql command is slower than breaking it up into multiple commands.

garland-culbreth committed:
split_table_insert: make cut_size an optional param

garland-culbreth committed:
split_table_insert: add test for split insertion

garland-culbreth committed:
split_table_insert: re-add blank line at eof

garland-culbreth committed:
split_table_insert: add documentation for cut_size in create_table.cpp

garland-culbreth committed:
Merge branch 'main' into feature/create-table-split-insert

garland-culbreth committed:
split_table_insert: contributor info and xrst docs

add my contributor info to the changed files and the contributor checker
add updates to the xrst docs in create_table.cpp to use the prototype xrst syntax

garland-culbreth committed:
feature/create-table-split-insert: temp

garland-culbreth committed:
split_table_insert: use clearer name for insertion size var

garland-culbreth committed:
split_table_insert: correct max_insert documentation
